zkvm/src/types.rs

```diff
@@ -182,6 +182,14 @@ impl Data {
         }
     }
 
+    /// Downcast the data item to a `Program` type.
+    pub fn to_program(self) -> Result<Vec<Instruction>, VMError> {
+        match self {
+            Data::Program(program) => Ok(program),
+            _ => Err(VMError::InconsistentWitness),
```

Comment:
Other downcast methods use `TypeNot<Type>`, let's do the same here?:

```suggestion
            _ => Err(VMError::TypeNotProgram),
```